homeassistant/components/smarttub/services.yaml

```diff
@@ -0,0 +1,22 @@
+set_primary_filtration:
```

Comment:
The service descriptions have changed and been extended, see: <https://developers.home-assistant.io/docs/dev_101_services/#service-descriptions>

Please update your services file accordingly.

Reply:
It looks like the new format can have an entity selector, which would be very useful here since there's usually only one applicable entity. But I don't see a good way to do it:

https://www.home-assistant.io/docs/blueprint/selectors/#configuration-variables

Integration, domain, and device class filters all seem too broad. Is there a way to limit the available entities to only the type which accepts this service call?

Reply:
I made a first pass.

The docs say to omit `entity_id` from the fields in `services.yaml` when there is a target selector, but then in the service tool "fill example data" doesn't add the `entity_id` attribute and the user has to add it manually. Is this normal?

![image](https://user-images.githubusercontent.com/436911/109038369-84276d80-7680-11eb-94fa-8ca542ec19d1.png)